lib/datadog/core/runtime/metrics.rb

```diff
@@ -78,6 +79,9 @@ def flush
               )
             end
           end
+
+          # Only on Ruby >= 3.2
+          try_flush { yjit_metrics.each { |metric, value| gauge(metric, value) } if Core::Environment::YJIT.available? }
```

Comment:
I think this explicitly declare all the values that it reports, which today would be:
```ruby
[:inline_code_size,
 :outlined_code_size,
 :freed_page_count,
 :freed_code_size,
 :live_page_count,
 :code_gc_count,
 :code_region_size,
 :object_shape_count]
```

Reply:
I've changed the iteration to explicit methods. The comments for each method I lifted directly from the Y[JIT source code](https://github.com/ruby/ruby/blob/2759cf46f28b1f5ba52f6b7686fd23cb8d792ede/yjit/src/stats.rs#L453).